api/ocm/internal/resolver.go

```diff
@@ -18,22 +18,139 @@ import (
 
 ////////////////////////////////////////////////////////////////////////////////
 
-type ResolverRule struct {
+// ComponentVersionResolver describes an object able
+// to map component version identities to a component version access,
+// the representtaion of the component version stored in a dedicated
+// OCM repository.
+// Such a resolver might optionally implement the ComponentResolver
+// interface to provide information about potential locations for
+// a particular component.
+type ComponentVersionResolver interface {
+	LookupComponentVersion(name string, version string) (ComponentVersionAccess, error)
+}
+
+// ResolvedComponentProvider is an interface for an
+// object providing optional access to a repository
+// and a provider for component versions.
+// Direct operations and operations of provided objects
+// may fail, if an optional underlying object is already closed
+// Therefore, it does not need to be explicitly closed.
+//
+// Be careful: Objects of this plain type should only be held as long as the
+// providing object is valid and not closed.
+type ResolvedComponentProvider interface {
+	Repository() (Repository, error)
+	LookupComponent(name string) (ResolvedComponentVersionProvider, error)
+}
+
+// ResolvedComponentVersionProvider is an interface for an
+// object providing access component versions.
+// Operations may fail, if an underlying object is already closed.
+//
+// Be careful: Objects of this plain type should only be held as long as the
+// providing object is valid and not closed.
+type ResolvedComponentVersionProvider interface {
+	GetName() string
+	LookupVersion(version string) (ComponentVersionAccess, error)
+	ListVersions() ([]string, error)
+}
+
+// ComponentResolver is an optional interface for a ComponentVersionResolver,
+// which can offer information about potential providers usable to lookup
+// versions for a dedicated component.
+//
+// Be careful, provided objects may refer to closable objects
+// held by the interface provider. They should only be used as long
+// as the object behind the ComponentResolver is not closed.
+type ComponentResolver interface {
+	LookupComponentProviders(name string) []ResolvedComponentProvider
+}
```

Comment:
I love the documentation, great! But it's in the `internal` package... :-/ 
As an API user, where would I see this doc then?

Reply:
Good question, unfortunately Go does not  forward the documentation to all type aliases, but you can follow the links provided for a type alias.